server/monitor.go

```diff
@@ -2337,6 +2338,25 @@ func (s *Server) Jsz(opts *JSzOptions) (*JSInfo, error) {
 	if opts.Streams {
 		opts.Accounts = true
 	}
+
+	// Check if we want a response from the leader only.
+	if opts.LeaderOnly {
+		js, cc := s.getJetStreamCluster()
+		if js == nil {
+			// Ignore
+			return nil, ErrJetStreamNotEnabled
```

Comment:
Here and on line 2355 if we return Err the error is sent back to the clients, meaning the !leaders still respond but in error.

These 2 lines should be `return nil, nil`

Reply:
Ugh, not even ... will dig deeper

Reply:
https://github.com/nats-io/nats-server/pull/1902